nengo/tests/test_cache.py

```diff
@@ -180,6 +181,19 @@ def test_corrupted_decoder_cache_index(tmpdir):
     assert len(os.listdir(cache_dir)) == 2  # index, index.lock
 
 
+def test_too_new_decoder_cache_index(tmpdir):
+    cache_dir = str(tmpdir)
+
+    # Write index with super large version numbers
+    with open(os.path.join(cache_dir, CacheIndex._INDEX), "wb") as f:
+        pickle.dump((1000, 1000), f)
```

Comment:
Could use `(CacheIndex.VERSION + 1, pickle.HIGHEST_PROTOCOL + 1)` to make it perfectly clear that these number are supposed to be higher than the current version numbers and avoid the "magic" numbers (but this is not a blocker to me).